### PR TITLE
[cpuinfo] fix arm64-osx build

### LINKED ISSUE #28271: cpuinfo[core,tools]:arm64-osx -> 2022-07-19
Package: cpuinfo[core,tools]:arm64-osx -> 2022-07-19

**Host Environment**

- Host: arm64-osx
- Compiler: AppleClang 14.0.0.14000029
-    vcpkg-tool version: 2999-12-31-unknownhash
    vcpkg-scripts version: 7643561c2 2022-12-08 (24 hours ago)

**Failure logs**

```
-- Using cached pytorch-cpuinfo-5e63739504f0f8e18e941bd63b2d6d42536c7d90.tar.gz.
-- Extracting source /Users/leanderSchulten/git_projekte/Lichtsteuerung/vcpkg/downloads/pytorch-cpuinfo-5e63739504f0f8e18e941bd63b2d6d42536c7d90.tar.gz
-- Using source at /Users/leanderSchulten/git_projekte/Lichtsteuerung/vcpkg/buildtrees/cpuinfo/src/42536c7d90-2a15ca16d1.clean
-- Configuring arm64-osx
-- Building arm64-osx-dbg
-- Building arm64-osx-rel
-- Fixing pkgconfig file: /Users/leanderSchulten/git_projekte/Lichtsteuerung/vcpkg/packages/cpuinfo_arm64-osx/lib/pkgconfig/libcpuinfo.pc
-- Fixing pkgconfig file: /Users/leanderSchulten/git_projekte/Lichtsteuerung/vcpkg/packages/cpuinfo_arm64-osx/debug/lib/pkgconfig/libcpuinfo.pc
CMake Error at scripts/cmake/vcpkg_copy_tools.cmake:32 (message):
  Couldn't find tool "cpuid-dump":

      neither "/Users/leanderSchulten/git_projekte/Lichtsteuerung/vcpkg/packages/cpuinfo_arm64-osx/bin/cpuid-dump" nor "/Users/leanderSchulten/git_projekte/Lichtsteuerung/vcpkg/packages/cpuinfo_arm64-osx/bin/cpuid-dump.app" exists
Call Stack (most recent call first):
  ports/cpuinfo/portfile.cmake:53 (vcpkg_copy_tools)
  scripts/ports.cmake:147 (include)



```

### PR DESCRIPTION
**Pull request by autoantwort**
Closes #28271

### CODE REVIEW

File: ports/cpuinfo/portfile.cmake [hunk 1]
```diff
@@ -50,8 +50,12 @@ vcpkg_fixup_pkgconfig() # pkg_check_modules(libcpuinfo)
 file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/include")
 
 if("tools" IN_LIST FEATURES)
+    set(additional_tools "")
+    if(EXISTS "${CURRENT_PACKAGES_DIR}/bin/cpuid-dump")
+        list(APPEND additional_tools "cpuid-dump")
```
Comment (autoantwort) [inline ports/cpuinfo/portfile.cmake:55]: The tool does not exists on arm64-osx

**Review by JonLiu1993 — APPROVED** [review]

**Comment by JavierMatosD** [comment]
Thank you!

File: ports/cpuinfo/portfile.cmake [hunk 2]
```diff
@@ -50,8 +50,12 @@ vcpkg_fixup_pkgconfig() # pkg_check_modules(libcpuinfo)
 file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/include")
 
 if("tools" IN_LIST FEATURES)
+    set(additional_tools "")
+    if(EXISTS "${CURRENT_PACKAGES_DIR}/bin/cpuid-dump")
```
Comment (dg0yt) [inline ports/cpuinfo/portfile.cmake:54]: Lack of `VCPKG_TARGET_EXECUTABLE_SUFFIX` probably causes #28452.

Comment (autoantwort) [reply ports/cpuinfo/portfile.cmake:54]: I have created https://github.com/microsoft/vcpkg/pull/28725 to fix it.

**Pull request merged**